web/courses.go

```diff
@@ -97,25 +97,22 @@ func (s *QuickFeedService) rejectEnrollment(ctx context.Context, sc scm.SCM, enr
 }
 
 // enrollStudent enrolls the given user as a student into the given course.
-func (s *QuickFeedService) enrollStudent(ctx context.Context, sc scm.SCM, enrolled *qf.Enrollment) error {
+func (s *QuickFeedService) enrollStudent(ctx context.Context, sc scm.SCM, query *qf.Enrollment) error {
 	// course and user are both preloaded, no need to query the database
-	course, user := enrolled.GetCourse(), enrolled.GetUser()
+	course, user := query.GetCourse(), query.GetUser()
 
 	// check whether user repo already exists,
 	// which could happen if accepting a previously rejected student
 	repo, err := s.getRepo(course, user.GetID(), qf.Repository_USER)
 	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
 		return fmt.Errorf("failed to get %s repository for %q: %w", course.Code, user.Login, err)
 	}
-
+	// Use enrollment with full updated info to ensure that gorm Select.Updates works correctly.
+	query.Status = qf.Enrollment_STUDENT
 	s.logger.Debugf("Enrolling student %q in %s; has database repo: %t", user.Login, course.Code, repo != nil)
 	if repo != nil {
 		// repo already exist, update enrollment in database
-		return s.db.UpdateEnrollment(&qf.Enrollment{
-			UserID:   user.ID,
-			CourseID: course.ID,
-			Status:   qf.Enrollment_STUDENT,
```

Comment:
If we want to only update some fields in the database records we have to use gorm's `Select("Field1", "Field2")` and this is something we would like to avoid as these field names are easy to forget in case of changes, am I correct?